File: Lab6/Store.java
import java.util.*;

public class Store {
    private CashRegister cashRegister;
    private LinkedList<Product> products = new LinkedList<Product>();

    public Store() {
        cashRegister = new CashRegister();
        products.add(new Product("Whiteboard Marker", 85, 1.50));
        products.add(new Product("Whiteboard Eraser", 45, 5.00));
        products.add(new Product("Black Pen", 100, 1.50));
        products.add(new Product("Red Pen", 100, 1.50));
        products.add(new Product("Blue Pen", 100, 1.50));
    }

    private char readChoice() {
        return In.nextChar();
    }

    private void sell() {
        // Implementation of sell method
    }

    private void restock() {
        // Implementation of restock method
    }

    private void viewStock() {
        for (Product product : products) {
            System.out.println(product);
        }
    }

    private void viewCash() {
        System.out.println(cashRegister);
    }

    private void pruneProducts() {
        products.removeIf(Product::isEmpty);
    }

    private String readName() {
        return In.nextLine();
    }

    private double readPrice() {
        return In.nextDouble();
    }

    private int readNumber() {
        return In.nextInt();
    }

    private void help() {
        System.out.println("Menu options");
        System.out.println("s = sell");
        System.out.println("r = restock");
        System.out.println("v = view stock");
        System.out.println("c = view cash");
        System.out.println("p = prune products");
        System.out.println("x = exit");
    }

    public static void main(String[] args) {
        // Main implementation
    }
}

File: Lab6/Store.py
from Product import Product
from CashRegister import CashRegister

class Store:
    def __init__(self):
        self.cash_register = CashRegister()
        self.products = [
            Product("Whiteboard Marker", 85, 1.50),
            Product("Whiteboard Eraser", 45, 5.00),
            Product("Black Pen", 100, 1.50),
            Product("Red Pen", 100, 1.50),
            Product("Blue Pen", 100, 1.50)
        ]

    def sell(self):
        # Implementation of sell method
        pass

    def restock(self):
        # Implementation of restock method
        pass

    def view_stock(self):
        for product in self.products:
            print(product)

    def view_cash(self):
        print(self.cash_register)

    def prune_products(self):
        self.products = [product for product in self.products if not product.is_empty()]

    def help(self):
        print("Menu options")
        print("s = sell")
        print("r = restock")
        print("v = view stock")
        print("c = view cash")
        print("p = prune products")
        print("x = exit")

    @staticmethod
    def main():
        # Main implementation
        pass

if __name__ == "__main__":
    Store.main()

